### DISCUSSION TITLE
WordPress: Akeeba\Panopticon\Exception\SiteConnection\WebServicesInstallerNotEnabled

### DISCUSSION THREAD

**Discussion by rvalitov** [akeeba/panopticon#804] [Q&A (Community Support)] [answered]
Hi! When I try to add a WP website, I get error:
```
Could not connect to the remote site.

 Troubleshooting Information
Uh oh. Something is totally broken here…

Normally we can detect the nature of the connection error and provide some useful troubleshooting information here, or at least point you to some kind of useful, third party troubleshooting resource. When this is not the case, we can at least print out an error message which will help us help you troubleshoot it.

Instead, we got an error which we should be able to provide troubleshooting information for (Akeeba\Panopticon\Exception\SiteConnection\WebServicesInstallerNotEnabled), but we don't have the code for it. This should have not happened.

Please delete all files and folders inside the tmp folder of Akeeba Panopticon. This addresses the problem of this page's cached copy being severely out of date. If this doesn't help, please let us know.
```
Some info:

- [x] I have no issues with Joomla websites
- [x] I use the latest version of Panopticon and connector plugin. 
- [x] The `wp-json`  URL works and shows the `routes` section.
- [x] No firewall is installed

**Comment by nikosdion** [comment]
Have you selected **WordPress** as the site type in Panopticon?

**Reply by rvalitov** [reply]
Sure. After enabling debug mode in Panopticon, I see error
```json
{"code":"rest_no_route","message":"A suitable route for the URL and request method was not found.","data":{"status":404}}

```

**Reply by rvalitov** [reply]
![image](https://github.com/user-attachments/assets/62e0ec00-faf3-4d32-a808-56bd0d6d6cae)

**Reply by nikosdion** [reply]
Based on the information on this message, you have not installed or enabled the Akeeba Panopticon connector plugin on your site. That's what the `rest_no_route` error boils down to. It says that WordPress does not know how to route the WordPress JSON API URL we are giving it. This URL route is registered by our plugin. So the plugin is not installed, or not activated, or a third party plugin (e.g. a security plugin) is screwing up the WordPress REST API.

**Reply by rvalitov** [reply, marked as answer]
I had WP 5.3.18. Had to upgrade at least to 5.6.14, otherwise couldn't add the website to Panopticon. The connector showed no issues or minimal requirements warning. I had no firewall or security plugins installed/enabled.

**Reply by nikosdion** [reply]
The problem is that WordPress' REST API is getting endpoints added, but the minimum WordPress version for each one is not even mentioned in [its documentation](https://developer.wordpress.org/rest-api/reference). The endpoint I was using to test the connection, `wp/v2/plugins`, was apparently not present in 5.3. I have no idea when it was added. I would have to install all versions from 5.0 all the way to 5.6.14 to see when it was added. That's... insane.

Instead, I installed 5.0 and checked which endpoints are available. After all, I only need an endpoint to test if we can connect to the site, not to read any data. Apparently, `wp/v2/posts` exists in 5.0 so this is what I will be using from now on.
